be/src/service/staros_worker.cpp

```diff
@@ -73,13 +73,32 @@ void shutdown_staros_worker() {
     g_worker = nullptr;
 }
 
-StatusOr<staros::starlet::ShardInfo> get_shard_info(staros::starlet::ShardId shard_id) {
+static const char* const kStarletPrefix = "staros_";
+StatusOr<std::string> get_staros_shard_path(uint64_t shard_id) {
     if (g_worker == nullptr) {
         return Status::InternalError("init_staros_worker() must be called before get_shard_info()");
     }
-    return g_worker->get_shard_info(shard_id);
+    ASSIGN_OR_RETURN(auto shardinfo, g_worker->get_shard_info(shard_id));
+    return fmt::format("{}{}", kStarletPrefix, shardinfo.obj_store_info.uri);
 }
 
 } // namespace starrocks
+#else // USE_STAROS
+#include <string>
 
+#include "common/statusor.h"
+namespace starrocks {
+void set_tabletmanager_test_dir(std::string dir) {
+    kTabletManagerTestDir = dir;
+};
+void clear_tabletmanager_test_dir() {
+    kTabletManagerTestDir = "";
+}
+StatusOr<std::string> get_staros_shard_path(uint64_t shard_id) {
+    if (kTabletManagerTestDir == "") {
+        return Status::InternalError("please set TabletManagerTestDir");
+    }
+    return kTabletManagerTestDir;
+};
```

Comment:
Remove these functions and add a derived class of `GroupAssigner` in the unit test and use the derived class to mock the group path.